tests/helper/sshclient.py

```diff
@@ -89,6 +93,17 @@ def __init__(
         self.ssh_key_filepath = path.expanduser(sshconfig['ssh_key_filepath'])
 
 
+    def _increase_retry_count(self, wait: bool = True):
+        if self.retry_count >= self.ssh_max_retries:
+            max_timeout = self.retry_count * self.ssh_retry_timeout_seconds
+            # FIXME: this should not be pytest.exit() as it will exit immediately without cleanup
+            pytest.exit(f"Unable to establish an SSH connection after {max_timeout=} seconds. Aborting all tests.", returncode=5) 
```

Comment:
Nit pick: do we need to support the case where `wait` param is set to false? 

In the wait=False case we would not do a `time.sleep(self.ssh_retry_timeout_seconds)`, but after `self.retry_count >= self.ssh_max_retries` retries, we exit with the message that we have waited for `max_timeout`

Reply:
If we remove the wait param, we should reconsider the name. 

before it was clear that this function has the side effect of a sleep. But without a param I would prefer to have it in the function name e.g.`_increase_retry_with_wait`